src/createExplorer.js

```diff
@@ -39,43 +57,66 @@ module.exports = function createExplorer(options) {
         return fileCache.get(absoluteConfigPath);
       }
 
-      const load =
-        path.basename(absoluteConfigPath) === 'package.json'
-          ? () => loadPackageProp(path.dirname(absoluteConfigPath), options)
-          : () => loadDefinedFile(absoluteConfigPath, options);
+      let load;
```

Comment:
If you don't mind, could you explain this change?

Reply:
Any particular part of the change?

The change from a ternary to a full if-else block was because I'd prefer not to use ternaries when the different values can be multiple lines. Creating new objects instead of passing the `options` object directly pleases Flow because it knows that nobody is going to mutate the values somewhere else — and doing that made the functions take up a few lines.

Reply:
The code earlier was not using `packageProp`. So now if `absoluteConfigPath` is for a `package.json` and `packageProp` is `false`, we return the entire file as the result?

Reply:
Yeah — I think this is one case where Flow typing revealed a bug, right? The code earlier was *implicitly* using `packageProp` by passing `options`. Before, we would have tried to access a property in `package.json` that didn't exist, which would error.

Another possible move here would be to throw an error if `configPath` is `package.json` and `packageProp` is not defined.

Reply:
Added a clear error message and tests to ensure it comes through.